### DISCUSSION TITLE
How to setup the dashboard? (localhost version)

### DISCUSSION THREAD

**Discussion by Androz2091** [Androz2091/AtlantaBot#384] [Q&A] [answered]
How to setup the dashboard to make it run on my computer locally (without a domain)?

**Comment by Androz2091** [comment, marked as answer]
## Hello :wave:

Here is a step-by-step tutorial to learn how to launch the dashboard on your computer. If you want to use a custom domain name (like `dashboard.my-awesome-bot.com`), use **[this tutorial](https://github.com/Androz2091/AtlantaBot/discussions/371)** instead. Here the URL will be `http://localhost:8080` and only you will be able to access it.

# Step 1 : edit the config file 🔧

So, first of all, edit the dashboard part of your config file so it looks like this:

```js
dashboard: {
		enabled: true,
		baseURL: "http://localhost:8080",
		port: 8080,
		failureURL: "https://atlanta-bot.fr",

        // read below to learn how to edit these values
		logs: "",
		secret: "",
		expressSessionPassword: "",
},
```

* **secret** should be your client secret. It can be found at https://discord.dev in your application settings.
* **expressSessionPassword** should be a totally random password. (for example `IL0v3Atlanta`)
* **logs** should be the ID of a channel of your server, where the logs will be sent.

Once your config file is updated, restart your bot.

# Step 2 : configure your discord app

This is the last step of the tutorial, and it is really easy. Go to your discord developer dashboard (https://discord.dev) and edit your application settings.

* Go to `OAuth2` tab.
* Add a redirect URL with the following content: `http://localhost:8080/api/callback`).

# Congratulations, you did it :tada:

Congratulations, you now have your dashboard hosted on your computer! You can access it here: `http://localhost:8080`! If you want everyone to access it, follow **[this tutorial](https://github.com/Androz2091/AtlantaBot/discussions/371)**!

If you need more help, we are available on Discord to help you https://www.atlanta-bot.fr/discord. And if you want more tutorials you can add a star to this repository :star:

Add a star ⭐  
React with 👍  to support

**Reply by lambor590** [reply]
Thank you so much

**Reply by joeyk710** [reply]
When I try to set up the dashboard locally and go to localhost:8080, the discord sign-in fails and says "Invalid OAuth2 redirect_ui". Is there any way to fix this?

**Reply by Androz2091** [reply]
> When I try to set up the dashboard locally and go to localhost:8080, the discord sign-in fails and says "Invalid OAuth2 redirect_ui". Is there any way to fix this?

You maybe have missing something in Step 2. Can you retry?

**Reply by Marwinzien** [reply]
In "baseURL", we need ":8080" after "localhost" :)

**Reply by Ryan0204** [reply]
![image](https://user-images.githubusercontent.com/82715592/116529256-545e4580-a90f-11eb-979b-a93772ebff95.png)
Me too.  But i already add a redirect URL with the following content: http://localhost:8080/api/callback

**Reply by Ryan0204** [reply]
Any discord support server?

**Reply by iNOEEL** [reply]
Hi, i would like to point out that the dashboard config is a bit incomplete, since for baseURL the port must also be specified for localhost.

like:
baseURL: "http://localhost:8080",

otherwise everything else works perfectly, excellent!

**Reply by Androz2091** [reply]
@iNOEEL edited, thank you for your report 👍

**Reply by Aayush-683** [reply]
im stuck in a discord login loop using local host, when i log in, it reloads and then asks again for authorisation..

**Reply by minster586** [reply]
same here @BlackKnight683

**Reply by PHPxCODER** [reply]
> im stuck in a discord login loop using local host, when i log in, it reloads and then asks again for authorisation..

aah you need to paste your client secret for that to work.

**Comment by dev-odf** [comment]
What can we do if we host it on pterodactyle

**Comment by hotsu0p** [comment]
hello @Androz2091 the code doesnt work anymore
